sdk/storage/storage-blob-changefeed/src/utils/utils.common.ts

```diff
@@ -0,0 +1,105 @@
+import { URLBuilder } from "@azure/core-http";
+import { ContainerClient } from "@azure/storage-blob";
+import { CHANGE_FEED_SEGMENT_PREFIX, CHANGE_FEED_INITIALIZATION_SEGMENT } from "./constants";
+
+const millisecondsInAnHour = 60 * 60 * 1000;
+export function ceilToNearestHour(date: Date | undefined): Date | undefined {
+  if (date === undefined) {
+    return undefined;
+  }
+  return new Date(Math.ceil(date.getTime() / millisecondsInAnHour) * millisecondsInAnHour);
+}
+
+export function floorToNearestHour(date: Date | undefined): Date | undefined {
+  if (date === undefined) {
+    return undefined;
+  }
+  return new Date(Math.floor(date.getTime() / millisecondsInAnHour) * millisecondsInAnHour);
+}
+
+/**
+ * Get URI from an URL string.
+ *
+ * @export
+ * @param {string} url Source URL string
+ * @returns {(string | undefined)}
+ */
+export function getURI(url: string): string | undefined {
+  const urlParsed = URLBuilder.parse(url);
+  return `${urlParsed.getHost()}${urlParsed.getPort()}${urlParsed.getPath()}`;
```

Comment:
The templated string will probably never be `undefined`?

Reply:
Good catch.